homeassistant/components/sensor/folder.py

```diff
@@ -58,21 +66,45 @@ class Folder(Entity):
 
     ICON = 'mdi:folder'
 
-    def __init__(self, folder_path, filter_term):
+    def __init__(self, folder_path, filter_term, hass):
         """Initialize the data object."""
         folder_path = os.path.join(folder_path, '')  # If no trailing / add it
         self._folder_path = folder_path   # Need to check its a valid path
         self._filter_term = filter_term
+        self._hass = hass
```

Comment:
No need to pass in `hass`. `add_devices` will already set the `self.hass` for you after it has been added.

Reply:
awesome